api/enki.py

```diff
@@ -190,39 +162,56 @@ def count_loans_and_holds(patron, pin):
             )
 
     def request(self, url, method='get', extra_headers={}, data=None,
-                params=None, exception_on_401=False):
-        """Make an HTTP request, acquiring/refreshing a bearer token
-        if necessary.
-        """
+                params=None, retry_on_timeout=True, **kwargs):
+        """Make an HTTP request to the Enki API."""
         headers = dict(extra_headers)
-        if exception_on_401:
-            disallowed_response_codes = ["401"]
-        else:
-            disallowed_response_codes = None
-            response = self._make_request(
-                url=url, method=method, headers=headers,
-                data=data, params=params,
-                disallowed_response_codes=disallowed_response_codes
+        try:
+            return self._request(
+                method, url, headers=headers, data=data,
+                params=params, timeout=90,
+                disallowed_response_codes=None,
+                **kwargs
+            )
+        except RequestTimedOut, e:
+            if not retry_on_timeout:
+                raise e
+            self.log.info(
+                "Request to %s timed out once. Trying a second time.", url
             )
-        if response.status_code == 401:
-            # This must be our first 401, since our second 401 will
-            # make _make_request raise a RemoteIntegrationException.
-            #
-            # The token has expired. Get a new token and try again.
-            self.token = None
             return self.request(
-                url=url, method=method, extra_headers=extra_headers,
-                data=data, params=params, exception_on_401=True
+                url, method, extra_headers,
+                data, params, retry_on_timeout=False,
+                **kwargs
             )
-        else:
-            return response
 
+    def _request(self, method, url, headers, data, params, **kwargs):
+        """Actually make an HTTP request.
+
+        MockEnkiAPI overrides this method.
+        """
+        return HTTP.request_with_timeout(
+            method, url, headers=headers, data=data,
+            params=params, timeout=90, disallowed_response_codes=None,
```

Comment:
Doesn't matter much, but `timeout` and `disallowed_response_codes` are in the kwargs too.

Reply:
Thanks, it turns out this does matter a lot.